### PR TITLE
remake tokens

### PR DESCRIPTION
**Pull request by VladimirAfanasievFS**
# Summary

#### A brief description of the pull request.

#5 

#### Screenshots in case of UI changes

# Test plan

#### List of steps to manually test introduced functionality:

# Review notes

#### While reviewing pull-request (especially when it's your pull-request), please make sure that:

- you named PR close to issue name and linked that issue
- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- required ENV variables added and described in `.env.example`
- pull request is shorter than 300 lines

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,7 +64,7 @@
     "storybook-formik": "^2.4.2",
     "styled-components": "^5.3.5",
     "ts-node": "^10.9.1",
-    "typescript": "^4.8.2",
+    "typescript": "^5.0.4",
```
Comment (IlgizGilmanov) [inline package.json:67]: Not a PR question: is there any reason to setup typescript as a prod dependency? There're recommendations to install it as a dev dependency https://www.typescriptlang.org/download
image

Comment (IlgizGilmanov) [inline package.json:67]: When running `yarn install`: 
image

File: lib/apollo/types.ts [hunk 2]
```diff
@@ -1,9 +1,7 @@
...
+import { NextApiRequest, NextApiResponse, NextComponentType, NextPageContext } from 'next';
 import { AppContext, AppProps } from 'next/app';
+import { ParsedUrlQuery } from 'querystring';
 import { ApolloClient, NormalizedCacheObject } from '@apollo/client';
-import { NextApiRequest, NextApiResponse, NextComponentType, NextPageContext } from 'next';
-
-import { AccessTokenManager } from 'lib/auth/withAccessTokenManager';
 
 export type TToken = string | null;
 export type TExpires = number | null;
```
Comment (IlgizGilmanov) [inline lib/apollo/types.ts:7]: Can be removed as I understand

File: package.json (hunk 1 above)
Comment (VladimirAfanasievFS) [reply package.json:67]: I agree, no reason. Fixes

File: lib/apollo/apolloLinks.ts [hunk 3]
```diff
@@ -1,8 +1,15 @@
 /* eslint-disable no-console */
-import { ApolloLink, fromPromise } from '@apollo/client';
+import Router from 'next/router';
+import { TokenRefreshLink } from 'apollo-link-token-refresh';
+import Cookie from 'universal-cookie';
+import { ApolloLink } from '@apollo/client';
 import { onError } from '@apollo/client/link/error';
+
 import isSSR from 'config/isSSR';
```
[the remark this answers is not in the document]
Comment (VladimirAfanasievFS) [reply lib/apollo/apolloLinks.ts]: Changed

File: lib/apollo/types.ts (hunk 2 above)
Comment (VladimirAfanasievFS) [reply lib/apollo/types.ts]: +

File: __tests__/helpers/renderWithApolloClient.tsx [hunk 4]
```diff
@@ -1,14 +1,7 @@
-import { ReactElement } from 'react';
-import { TPageProps } from 'lib/apollo/types';
-import { ApolloClient, HttpLink, InMemoryCache, ApolloProvider } from '@apollo/client';
+import React, { ReactElement } from 'react';
```
Comment (IlgizGilmanov) [inline __tests__/helpers/renderWithApolloClient.tsx:1]: We can remove `React` from here

File: lib/apollo/hooks/state/meActivity/meActivity.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { filterAvailableNodes } from 'helpers';
+import { useMeActivitiesQuery } from 'graphql/queries/__generated__/meActivities.generated';
+import { Activity, ActivityEvent } from 'graphql/types';
+import { TActivity } from './types';
+
+export const useMeActivities = ({ before, after, event, pageSize = 5 }: TActivity) => {
+  const { loading, error, data } = useMeActivitiesQuery({
+    variables: {
+      events: event ? [event] : Object.values(ActivityEvent),
```
Comment (IlgizGilmanov) [inline lib/apollo/hooks/state/meActivity/meActivity.ts:9]: Why we don't use `events` in the `useActivities`?

Comment (VladimirAfanasievFS) [reply lib/apollo/hooks/state/meActivity/meActivity.ts:9]: It is different query, public and private. With different signature. 
image
image

File: components/pages/activity/constants.ts [hunk 6]
```diff
@@ -1,9 +1,9 @@
 import { ActivityEvent } from 'graphql/types';
-import { TActivityOption, TAvtivitySizeOption } from './types';
+import { TActivityOption, TActivitySizeOption } from './types';
 
-export const ACTIVITY_PAGE_SIZES = [5, 10, 25, 50];
+export const ACTIVITY_PAGE_SIZES = [3, 6, 9, 12];
```
Comment (IlgizGilmanov) [inline components/pages/activity/constants.ts:4]: Maybe just `[5, 10, 15]`?)

Comment (VladimirAfanasievFS) [reply components/pages/activity/constants.ts:4]: 15 calls depth error((

Comment (IlgizGilmanov) [reply components/pages/activity/constants.ts:4]: ((

**Review by IlgizGilmanov — APPROVED** [review]
:thumbsup:

**Pull request merged**